habitat/tasks/rearrange/sub_tasks/nav_to_obj_task.py

```diff
@@ -0,0 +1,307 @@
+                    self._nav_target_angle,
+                )
+                self.cache.save(self.start_states)
+            start_pos, start_rot = get_robo_start_pos(
+                sim, self._nav_target_pos
+            )
+        else:
+            if (
+                episode_id in self.start_states
+                and not self._config.FORCE_REGENERATE
+            ):
+                (
+                    self._nav_target_pos,
+                    self._nav_target_angle,
+                    start_pos,
+                    start_rot,
+                ) = self.start_states[episode_id]
+
+                sim.robot.base_pos = mn.Vector3(
+                    start_pos[0],
+                    sim.robot.base_pos[1],
+                    start_pos[2],
+                )
+                sim.robot.base_rot = start_rot
+            else:
+                (
+                    self._nav_target_pos,
+                    self._nav_target_angle,
+                    start_pos,
+                    start_rot,
+                ) = self._generate_nav_start_goal(episode)
+                self.start_states[episode_id] = (
+                    self._nav_target_pos,
+                    self._nav_target_angle,
+                    start_pos,
+                    start_rot,
+                )
+                self.cache.save(self.start_states)
+
+            targ_idxs, goal_pos = sim.get_targets()
+
+        observations = super().reset(episode)
+
+        if not sim.pathfinder.is_navigable(self._nav_target_pos):
+            print("Goal is not navigable")
+
+        if self._config.DEBUG_GOAL_POINT:
+            sim.viz_ids["nav_targ_pos"] = sim.visualize_position(
+                self._nav_target_pos,
+                sim.viz_ids["nav_targ_pos"],
+                r=10.0,
+            )
+
+        return observations
+
+
+def get_nav_targ(
+    task_cls_name,
+    task_def_path,
+    config,
+    sim,
+    env,
+    dataset,
+    task_kwargs,
+    episode,
+):
+    orig_state = sim.capture_state(with_robot_js=True)
+    load_task_object(
+        task_cls_name,
+        task_def_path,
+        config,
+        env,
+        dataset,
+        True,
+        task_kwargs,
+        episode,
+    )
+    robo_pos = sim.robot.base_pos
+    heading_angle = sim.robot.base_rot
+
+    sim.set_state(orig_state, set_hold=True)
+
+    return robo_pos, heading_angle
+
+
+def get_nav_from_obj_to(nav_name, obj_to, sim):
+    task_args = {}
+
+    if nav_name.startswith("TARG"):
+        raise ValueError("Place task not supported yet")
+    else:
+        task_cls_name = "RearrangePickTask-v0"
+        task_args = {"obj": obj_to}
+        obj_id = sim.scene_obj_ids[obj_to]
+        rom = sim.get_rigid_object_manager()
+        abs_true_point = rom.get_object_by_id(
+            obj_id
+        ).transformation.translation
+
+    return abs_true_point, task_cls_name, task_args
+
+
+def get_robo_start_pos(sim, nav_targ_pos):
+    timeout_len = 1000
```

Comment:
I do not think this is a proper "timeout". I would rename to `max_iterations` for example. The code seems to be sampling a random position and then checking if it is navigable to the target. I think checking if the start and end positions are navigable should be in its own method (seems useful on its own).
I would also rename this method to reflect the randomness : `sample_transform_navigable_to`